packages/helpers/classes/personalization.js

```diff
@@ -275,43 +275,26 @@ class Personalization {
    * To JSON
    */
   toJSON() {
-
     //Get data from self
-    const {
-      to, cc, bcc, subject, headers, customArgs, sendAt,
-      substitutions, substitutionWrappers,
-    } = this;
+    const {to, cc, bcc, subject, headers, customArgs, sendAt, substitutions, substitutionWrappers} = this;
 
     //Initialize with mandatory values
     const json = {to};
 
     //Arrays
-    if (Array.isArray(cc) && cc.length > 0) {
-      json.cc = cc;
-    }
-    if (Array.isArray(bcc) && bcc.length > 0) {
-      json.bcc = bcc;
-    }
-
+    json.cc = Array.isArray(cc) && cc.length > 0 ? cc : null;
+    json.bcc = Array.isArray(bcc) && bcc.length > 0 ? bcc : null;
     //Objects
-    if (Object.keys(headers).length > 0) {
-      json.headers = headers;
-    }
+    json.headers = Object.keys(headers).length > 0 ? headers : null;
```

Comment:
We are getting a failing test due to these lines, because the helper is looking for anything that is set and passing it through. When the params are null, they are considered "set" so they attempt to be passed to the API. The API is rejecting, because these params are set but don't contain any data. 

```
1 failing
  1) sgMail.send() should send a basic email:
     Bad Request
  (400)
    undefined
      personalizations.[].cc
      undefined
    undefined
      personalizations.[].custom_args
      undefined
    undefined
      personalizations.[].headers
      undefined
    undefined
      personalizations.[].send_at
      undefined
    undefined
      personalizations.[].subject
      undefined
    undefined
      personalizations.[].bcc
      undefined
      at Request.http [as _callback] (packages/client/src/client.js:31:27)
      at endReadableNT (_stream_readable.js:1056:12)
      at _combinedTickCallback (internal/process/next_tick.js:138:11)
      at process._tickDomainCallback (internal/process/next_tick.js:218:9)
``` 
from here: https://travis-ci.org/sendgrid/sendgrid-nodejs/jobs/294620691

I like ternary operators because they are very clear. However, this is making me wonder if we should update these specific params to not be included through the helper if they are null - so that we can keep our code simple AND protect the fact that null should mean "don't send this param!". 

cc: @thinkingserious and @adamreisnz for thoughts.

Reply:
@mbernier @TheWillG This is why I personally dislike ternary operators -- unless the code and logic is very simple. They are quite prone to logic errors and sometimes it's hard to see what's going on, or what the authors of the code intended.

In my opinion there is no need for this kind of refactoring, as it doesn't make the code more clear, and there is no real reason to make the code shorter. So I'd prefer clarity and simplicity in this case, and to just leave as is unless it adds meaningful functionality.

I'm also worried that this would open the floodgates to other PR's like this. Does changing all if statements in this file mean we need to go and do it in every file of the sendgrid library now? If so, why? If not, why not? I hope you get the point 😄

Reply:
@adamreisnz 
Just a thought
```javascript
json.headers = Object.keys(headers).length > 0 ? headers : null;
```
null and undefined arent the same thing. null is actually assigning a value to the variable. Shouldnt the null actually be an `''` or an undefined in case ternary operators are used.